tower_cli/conf.py

```diff
@@ -219,21 +229,27 @@ def __getattr__(self, key):
             except configparser.NoOptionError:
                 continue
 
-            # We have a value; it may or may not be a string, though, so
-            # try to return it as an int, float, or boolean (in that order)
-            # before falling back to the string value.
-            type_method = ('getint', 'getfloat', 'getboolean')
-            for tm in type_method:
-                try:
-                    value = getattr(parser, tm)('general', key)
+            # We have a value; try to get its type and return it accordingly
+            try:
+                if CONFIG_PARAM_TYPE[key] == (click.STRING or click.Choice):
+                    value = getattr(parser, configparser.ConfigParser.get.__name__)('general', key)
+                    break
+                elif CONFIG_PARAM_TYPE[key] == click.BOOL:
+                    value = getattr(parser, configparser.ConfigParser.getboolean.__name__)('general', key)
+                    break
+                elif CONFIG_PARAM_TYPE[key] == click.FLOAT:
+                    value = getattr(parser, configparser.ConfigParser.getfloat.__name__)('general', key)
                     break
-                except ValueError:
-                    pass
+                elif CONFIG_PARAM_TYPE[key] == click.INT:
+                    value = getattr(parser, configparser.ConfigParser.getint.__name__)('general', key)
+                    break
+            except ValueError:
+                click.secho('Value for %s is not in expected type' % key)
 
-            # Write the value to the cache, so we don't have to do this lookup
-            # logic on subsequent requests.
-            self._cache[key] = value
-            return self._cache[key]
+        # Write the value to the cache, so we don't have to do this lookup
+        # logic on subsequent requests.
+        self._cache[key] = value
+        return self._cache[key]
```

Comment:
Some of the tests are failing because of this change in indentation, because it looks like we were relying on the `continue` to avoid these lines in the event that a variable was not defined in any of the available parsers. Easiest solution for you might be to just re-indent them.

Reply:
@AlanCoding Yeah and I think I will also have to remove those breaks. Otherwise it wont ever write a value to cache if I re-indent it.

Reply:
oh, I missed that.

Previously the `break` statement was breaking out of the `for tm in type_method` loop, and now it's breaking out of the `for parser in self._parsers` loop. I agree you just can't keep using the break statement in the same way. Like you say, removing the break might work fine, because you're using `elif` for the other conditions.

Reply:
@AlanCoding Yeah. My bad. I should have run those integration tests locally. I have made the change and ran them locally, all green now. Thanks for your review.